pylast/__init__.py

```diff
@@ -131,6 +149,34 @@ def _deprecation_warning(message):
 
 XML_ILLEGAL = re.compile(RE_XML_ILLEGAL)
 
+# Python <=3.3 doesn't support create_default_context()
+# <2.7.9 and <3.2 never did any SSL verification
+# FIXME This can be removed after 2017-09 when 3.3 is no longer supported and
+# pypy3 uses 3.4 or later, see
+# https://en.wikipedia.org/wiki/CPython#Version_history
+if sys.version_info < (3, 3, float("inf")) and sys.version_info > (3, 2):
+    import certifi
+    SSL_CONTEXT = ssl.SSLContext(ssl.PROTOCOL_TLSv1)
+    SSL_CONTEXT.verify_mode = ssl.CERT_REQUIRED
+    SSL_CONTEXT.options |= ssl.OP_NO_COMPRESSION
+    # Compact version of Modern and Intermediate from
+    # https://wiki.mozilla.org/Security/Server_Side_TLS
+    # Create the cipher string
+    cipher_string = """
+    CHACHA20+EECDH@STRENGTH
+    AESGCM+EECDH@STRENGTH
+    ECDHE+SHA+AES+aRSA@STRENGTH
+    DHE+SHA+AES+aRSA@STRENGTH
+    """
+    cipher_string = ' '.join(cipher_string.split())
+    SSL_CONTEXT.set_ciphers(cipher_string)
+    SSL_CONTEXT.load_verify_locations(certifi.where())
+
+# Python >3.4 and >2.7.9 has sane defaults
+elif sys.version_info > (3, 4) or (
+        sys.version_info < (3, 0) and sys.version_info > (2, 7, 9)):
```

Comment:
Can this use `_can_use_ssl_securely()`?

The logic for `_can_use_ssl_securely()` is slightly different: 
```python
v > (3, 3) or (v < (3, 0) and v > (2, 7, 9))
```

Compared to here:
```python
v > (3, 4) or (v < (3, 0) and v > (2, 7, 9))
```

Are they intentionally different? They have similar comments:
> `>3.4 and >2.7.9 has sane defaults so use SSL there`

And:
> `Python >3.4 and >2.7.9 has sane defaults`

---

If we cannot reuse `_can_use_ssl_securely()`, let's use interval comparison:
```python
v > (3, 4) or ((2, 7, 9) < v < (3, 0))
```